### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by ssaviak**

### CODE REVIEW

**Review by vSainiuk — APPROVED** [review]
Marvelous 💫

File: src/groupByKey.ts [hunk 1]
```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const resObj: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const keyOfItem = item[key] as string;
```
Comment (DanilWeda) [inline src/groupByKey.ts:9]:
```suggestion
    const keyOfItem = String(item[key]);
```

File: src/groupByKey.ts [hunk 2]
```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const resObj: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
```
Comment (DanilWeda) [inline src/groupByKey.ts:8]: You can use reduce instead forEach. In this case you don't need to create resObj on 6 line.

**Review by DanilWeda — APPROVED** [review]
Good job! 
Check comments for future solutions.

File: src/groupByKey.ts (hunk 1 above)
Comment (ssaviak) [reply src/groupByKey.ts:9]: Thanks. i was looking for solution like this

**Pull request closed without merging**